BookingQueueSubscriber/BookingQueueSubscriber.Services/MessageHandlers/EndpointUpdatedHandler.cs

```diff
@@ -11,26 +13,60 @@ namespace BookingQueueSubscriber.Services.MessageHandlers
     public class EndpointUpdatedHandler : IMessageHandler<EndpointUpdatedIntegrationEvent>
     {
         private readonly IVideoApiService _videoApiService;
+        private readonly ILogger<EndpointUpdatedHandler> _logger;
+        private const int RetryLimit = 3;
+        private const int RetrySleep = 3000;
 
-        public EndpointUpdatedHandler(IVideoApiService videoApiService)
+        public EndpointUpdatedHandler(IVideoApiService videoApiService, ILogger<EndpointUpdatedHandler> logger)
         {
             _videoApiService = videoApiService;
+            _logger = logger;
         }
 
         public async Task HandleAsync(EndpointUpdatedIntegrationEvent eventMessage)
         {
-            var conference = await _videoApiService.GetConferenceByHearingRefId(eventMessage.HearingId);
             ParticipantDetailsResponse defenceAdvocate = null;
+            ConferenceDetailsResponse conference = null;
+
             if (!string.IsNullOrEmpty(eventMessage.DefenceAdvocate))
             {
-                defenceAdvocate = conference.Participants.Single(x => x.ContactEmail ==
-                    eventMessage.DefenceAdvocate);
+                for (var retry = 0; retry <= RetryLimit; retry++)
```

Comment:
There's a SonarCloud code smell about reducing the cognitive complexity. We could potentially fix that by moving this `for` block into a separate method, eg `private async Task GetDefenceAdvocate(Guid hearingId)`

And then call it with `defenceAdvocate = await GetDefenceAdvocate(eventMessage.HearingId);`

Reply:
I'll look to getting this into a separate method

Reply:
Moved the retry for loop to its own function to help get rid of the code smell